feat(curriculum): workshop debugging registration form updates (#66731)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-debugging-registration-form/69cc1fa5d38fdf308398f910.md

This groups all related logs under a single, collapsible section titled "Register User," making it easier to navigate and debug.

# --hints--

You should use `console.groupCollapsed("Register User")` before the `try` block.

```js
assert.match(
  code,
  /console\.groupCollapsed\((["'])Register User\1\);\s*try\s*\{/
);
```

You should use `console.groupEnd()` at the bottom of the `registerUser` function.

```js
assert.match(
  code,
  /console\.table\(\s*\[\s*user\s*\]\s*\)\s*;[\s\S]*?console\.groupEnd\(\)/
);
```

# --seed--

## --seed-contents--

```js
const nameInput = document.getElementById("name-input");
const ageInput = document.getElementById("age-input");
const btn = document.getElementById("submit-btn");
const result = document.getElementById("result");

console.clear();

console.dir(nameInput);
console.dir(ageInput);

btn.addEventListener("click", registerUser);

function registerUser() {
  console.log("Button clicked");
  
  const name = nameInput.value;
  const age = ageInput.value;

  console.log("Name:", name);
  console.log("Age:", age);
  
  --fcc-editable-region--

>   try {

Reviewer: ```suggestion
  --fcc-editable-region--
  
  try {
```

    if (name.trim() === "") {
      throw new Error("Name is required");
    }
    if (age.trim() === "") {
      throw new Error("Age is required");
    }
    if (!Number.isInteger(Number(age))) {
      throw new Error("Invalid value for age");
    }
    result.textContent = "Registration successful";
  } catch (error) {
    console.error(error.message);
    result.textContent = error.message;
  }

  const user = {
    name: name,
    age: age,
  };

>   console.table([user]);

Reviewer: is it part of the spirit of the workshop to log invalid values? or should maybe the execution stopped inside the `catch`?

Author: > is it part of the spirit of the workshop to log invalid values? or should maybe the execution stopped inside the `catch`?

ah, good point. I overlooked that part. I'll deal with it once I've updated the tests for steps 7, 9, 10, 11

> --fcc-editable-region--

Reviewer: ```suggestion
  --fcc-editable-region--
  
  try {
    if (name.trim() === "") {
      throw new Error("Name is required");
    }
    if (age.trim() === "") {
      throw new Error("Age is required");
    }
    if (!Number.isInteger(Number(age))) {
      throw new Error("Invalid value for age");
    }
    result.textContent = "Registration successful";
  } catch (error) {
    console.error(error.message);
    result.textContent = error.message;
  }

  const user = {
    name: name,
    age: age,
  };
  console.table([user]);
  
--fcc-editable-region--
```

---
id: 69cc1fa5d38fdf308398f910
title: Step 15
challengeType: 0
dashedName: step-15
---

# --description--

You've been logging a lot of information to the console. As your project grows, this can quickly become hard to read.

To keep things organized, you can group related logs together.

Use `console.groupCollapsed("Register User")` at the beginning of the `registerUser` function and `console.groupEnd()` at the end of the function.

This groups all related console output for a registration attempt under a single, collapsible section titled `"Register User"`.

# --hints--

You should use `console.groupCollapsed("Register User")` at the beginning of the `registerUser` function.

```js
assert.match(
  code,

>   /console\.groupCollapsed\((["'])Register User\1\);\s*console\.log\((["'])Button clicked\2\)/

Reviewer 2: This currently requires a trailing semicolon after `console.groupCollapsed("Register User")`, but the instruction does not require semicolons and semicolonless JavaScript should also pass.

```suggestion
  /console\.groupCollapsed\((["'])Register User\1\);?\s*console\.log\((["'])Button clicked\2\)/
```

> );

Reviewer 2: Minor: the regex requires `console.groupCollapsed()` to sit directly above `console.log("Button clicked")`. A blank line between them is fine, but a comment fails. Matching against `removeJSComments` handles that.

```suggestion
assert.match(
  __helpers.removeJSComments(code),
  /console\.groupCollapsed\((["'])Register User\1\);?\s*console\.log\((["'])Button clicked\2\)/
);
```

```

> You should use `console.groupEnd()` after the `catch` block.

Reviewer 2: The description asks for `console.groupEnd()` "at the end of the function", so the hint should describe it the same way.

```suggestion
You should use `console.groupEnd()` at the end of the `registerUser` function.
```

```js
assert.match(
  code,
  /catch\s*\(error\)\s*\{[\s\S]*?\}\s*console\.groupEnd\(\)/
);
```

# --seed--

## --seed-contents--

```js
const nameInput = document.getElementById("name-input");
const ageInput = document.getElementById("age-input");
const btn = document.getElementById("submit-btn");
const result = document.getElementById("result");

console.clear();

console.dir(nameInput);
console.dir(ageInput);

btn.addEventListener("click", registerUser);

function registerUser() {
  console.log("Button clicked");
  
  const name = nameInput.value;
  const age = ageInput.value;
  const ageNum = Number(age);

  console.log("Name:", name);
  console.log("Age:", age);
  
  --fcc-editable-region--
  
  try {
    if (name.trim() === "") {
      throw new Error("Name is required");
    }
    if (age.trim() === "") {
      throw new Error("Age is required");
    }
    if (!Number.isInteger(ageNum) || ageNum < 13) {
      throw new Error("Invalid value for age");
    }

    const user = {
      name: name,

>       age: age,

Reviewer: ```suggestion
      age: age
```

}

button:active {
  box-shadow: inset 0 0 5px rgba(0, 0, 0, 0.4);
}

#age-help {
  margin: 0.5rem 0 0;
  font-size: 0.875rem;
  color: #dfdfe2;
}

#result {
  min-height: 1.5rem;
  margin-top: 20px;
  font-weight: 600;
  text-align: center;
}
```

# --solutions--

```js
const nameInput = document.getElementById("name-input");
const ageInput = document.getElementById("age-input");
const btn = document.getElementById("submit-btn");
const result = document.getElementById("result");

console.clear();

console.dir(nameInput);
console.dir(ageInput);

btn.addEventListener("click", registerUser);

function registerUser() {
  console.log("Button clicked");
  
  const name = nameInput.value;
  const age = ageInput.value;
  const ageNum = Number(age);

  console.log("Name:", name);
  console.log("Age:", age);
  
  console.groupCollapsed("Register User");

  try {
    if (name.trim() === "") {
      throw new Error("Name is required");
    }
    if (age.trim() === "") {
      throw new Error("Age is required");
    }
    if (!Number.isInteger(ageNum) || ageNum < 13) {
      throw new Error("Invalid value for age");
    }

    const user = {
      name: name,

>       age: age,

Reviewer: ```suggestion
      age: age
```
